http/src/main/java/com/hazelcast/jet/contrib/http/HttpListenerSinkBuilder.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.contrib.http;
+
+import com.hazelcast.function.FunctionEx;
+import com.hazelcast.function.SupplierEx;
+import com.hazelcast.jet.contrib.http.impl.HttpListenerSinkContext;
+import com.hazelcast.jet.core.Processor;
+import com.hazelcast.jet.core.ProcessorSupplier;
+import com.hazelcast.jet.core.processor.SinkProcessors;
+import com.hazelcast.jet.impl.pipeline.SinkImpl;
+import com.hazelcast.jet.pipeline.Sink;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.net.ssl.SSLContext;
+import java.util.Objects;
+
+import static com.hazelcast.jet.core.ProcessorMetaSupplier.forceTotalParallelismOne;
+import static com.hazelcast.jet.impl.pipeline.SinkImpl.Type.TOTAL_PARALLELISM_ONE;
+
+/**
+ * See {@link HttpListenerSinks#builder()}.
+ *
+ * @param <T> the type of the pipeline item.
+ */
+public class HttpListenerSinkBuilder<T> {
+
+    /**
+     * Default port for HTTP Listener sink
+     */
+    public static final int DEFAULT_PORT = 8081;
+
+    /**
+     * Default path for HTTP Listener sink
+     */
+    public static final String DEFAULT_PATH = "/";
+
+    private static final int PORT_MAX = 0xFFFF;
+
+    private int port = DEFAULT_PORT;
```

Comment:
Shouldn't the Sink also have the `hostFn` function? This way we always bind to the member address, which has same issues josef pointed out for the Source.

Also with the host fn it is not possible to use the current form of `sinkAddress` in HttpListenerSinks, it should take this `hostFn` into consideration.